src/components/Toaster/__tests__/ToasterProvider.test.tsx

```diff
@@ -302,3 +303,113 @@ describe('api.has', () => {
         expect(providerAPI.has(toastProps.name)).toBe(false);
     });
 });
+
+describe('interaction with Modal while closing Toaster', () => {
+    const MODAL_CONTENT = 'qwerty-modal';
+
+    const ModalAPI = ({onMount}: {onMount: (openModalFun: () => void) => any}) => {
+        const [open, setOpenFlag] = React.useState(true);
+        const closeModal = () => setOpenFlag(false);
+
+        React.useEffect(() => {
+            onMount(() => setOpenFlag(true));
+        }, []);
+
+        return (
+            <Modal open={open} onOutsideClick={closeModal}>
+                {MODAL_CONTENT}
+            </Modal>
+        );
+    };
+
+    function setup() {
+        let providerAPI: undefined | ToasterPublicMethods;
+        let openModal: undefined | (() => void);
+
+        render(
+            <ToasterProvider>
+                <ToastAPI
+                    onMount={(api) => {
+                        providerAPI = api;
+                    }}
+                />
+                <ModalAPI
+                    onMount={(_openModal) => {
+                        openModal = _openModal;
+                    }}
+                />
+                <ToasterComponent />
+            </ToasterProvider>,
+        );
+
+        if (!providerAPI || !openModal) {
+            throw new Error('Failed to setup test');
+        }
+
+        return {providerAPI, openModal};
+    }
+
+    it('Toaster was opened after Modal', async () => {
+        const {providerAPI, openModal} = setup();
+
+        act(openModal);
+
+        const modal = await screen.findByText(MODAL_CONTENT);
+        expect(modal).toBeInTheDocument();
+
+        act(() => {
+            providerAPI.add({...toastProps, isClosable: true});
+        });
+
+        const toast = getToast();
+        expect(toast).toBeInTheDocument();
+
+        fireEvent.click(toast);
+
+        expect(modal).toBeInTheDocument();
+
+        // eslint-disable-next-line
+        const closeToastButton = toast.querySelector('button');
```

Comment:
use [`within()`](https://testing-library.com/docs/dom-testing-library/api-within/) to find button by role in toaster

Reply:
[034deb5](https://github.com/gravity-ui/uikit/pull/792/commits/034deb5d82aeb9ffa746d562910b6fa2b9ea1651)